### PR TITLE
Implement species clustering

### PR DESCRIPTION
**Pull request by jakobjpeters**
__There is a new dependency requirement (`distinctipy==1.3.4`).__

Notes:
- User option to select number of species doesn't make sense anymore
- Reproduction should check for same-species
- Read docstrings for `Species` for info about the clustering
- `darken_color` seems to change the color entirely, so it's not currently being used
- I'm not sure if @sam-baird can see the colors well
- Latency
    - Opening app (due to loading `scikit-learn`)
    - Starting simulation (due to initial clustering)
    - Interacting with GUI (due to simulation being ran)

### CODE REVIEW

**Comment by jakobjpeters** [comment]
- Restricting cells to contain zero or one organism also implements sexual reproduction for plants that are adjacent to each other

**Comment by sam-baird** [comment]
Thanks Jakob! I haven't worked with sklearn before, but to fix returning the labels of clusters in different orders, maybe we could track the previous ordering in the `Species` class and reorder `self.labels_colors` as needed when returning the updated dictionary.

I, too, have noticed `darken_color` not really working for some colors so I'm okay with just removing it for now. It also probably makes tracking the species more difficult with the tone of the color changing every few frames. 

I'm working on revamping the simulation GUI where there will now be organization into separate resizeable paned windows. Maybe the bottom panel could show a plot of the clusters, similar to demonstrated [here]( https://scikit-learn.org/stable/auto_examples/cluster/plot_mean_shift.html#sphx-glr-auto-examples-cluster-plot-mean-shift-py)

**Comment by jakobjpeters** [comment]
That's a great idea, and I implemented it using the `sklearn.pairwise_distances_argmin` function to identify which clusters are closest to the previous clusters and the map the colors accordingly. It should be stable for both increasing and decreasing the number of organisms. Let me know if you see it being unstable. 

Got it!

Your idea for the bottom panel is awesome! I fully agree and will implement that.
